larpandoracontent/LArHelpers/LArHitWidthHelper.cc

```diff
@@ -275,4 +275,47 @@ void LArHitWidthHelper::GetExtremalCoordinatesX(const ConstituentHitVector &cons
     }
 }
 
+//------------------------------------------------------------------------------------------------------------------------------------------
+
+CartesianVector LArHitWidthHelper::GetClosestPointToLine2D(const CartesianVector &lineStart, const CartesianVector &lineDirection, const CaloHit *const pCaloHit)
+{
+    const CartesianVector &hitPosition(pCaloHit->GetPositionVector());
+    
+    if (std::fabs(lineDirection.GetZ()) < std::numeric_limits<float>::epsilon())
+        return hitPosition;
+    
+    float xOnLine(lineStart.GetX());
+    if (std::fabs(lineDirection.GetX()) > std::numeric_limits<float>::epsilon())
+    {
+        const float gradient(lineDirection.GetZ() / lineDirection.GetX());
+        xOnLine = ((hitPosition.GetZ() - lineStart.GetZ()) / gradient) + lineStart.GetX();
```

Comment:
Can just use += here given the initialisation above, rather than calling lineStart.GetX() again.